src/foam/nanos/auth/AuthenticatedAgentJunctionDAO.java

```diff
@@ -0,0 +1,160 @@
+      return getDelegate().put_(x, junctionObj);
+    }
+
+    // Check agent or user to authorize the request as.
+    User authorizedUser = agent != null ? agent : user;
+
+    // Check junction object relation to authorized user.
+    boolean authorized = SafetyUtil.equals(junctionObj.getTargetId(), authorizedUser.getId());
+
+    if ( junctionObj != null && ! authorized ) {
+      throw new AuthorizationException("Unable to update junction.");
+    }
+
+    return getDelegate().put_(x, junctionObj);
+  }
+
+  /**
+   * Return junction object based on logic within userAgentAuthorization method.
+   */
+  @Override
+  public FObject find_(X x, Object id) {
+    UserUserJunction junctionObj = (UserUserJunction) getDelegate().find_(x, id);
+    
+    if ( junctionObj == null )
+        return null;
+
+    // Check global permissions
+    if ( userAgentAuthorization(x, junctionObj, GLOBAL_AGENT_JUNCTION_READ) ) {
+      return junctionObj;
+    }
+
+    return null;
+  }
+
+  /**
+   * Allow users to remove junction objects based on logic within userAgentAuthorization method.
+   */
+  @Override
+  public FObject remove_(X x, FObject obj) {
+    UserUserJunction junctionObj = (UserUserJunction) getDelegate().inX(x).find(obj);
+
+    if ( junctionObj == null )
+        return null;
+
+    if ( userAgentAuthorization(x, junctionObj, GLOBAL_AGENT_JUNCTION_DELETE) ) {
+      return getDelegate().remove_(x, junctionObj);
+    }
+
+    throw new AuthorizationException("Unable to remove object.");
+  }
+
+  @Override
+  public void removeAll_(X x, long skip, long limit, Comparator order, Predicate predicate) {
+    DAO dao = getFilteredDAO(x, GLOBAL_AGENT_JUNCTION_DELETE);
+
+    dao.removeAll_(x, skip, limit, order, predicate);
+  }
+
+  @Override
+  public Sink select_(X x, Sink sink, long skip, long limit, Comparator order, Predicate predicate) {
+    DAO dao = getFilteredDAO(x, GLOBAL_AGENT_JUNCTION_READ);
+
+    return dao.select_(x, sink, skip, limit, order, predicate);
+  }
+
+  /**
+   * Checks if agent is present within the context, if so check id on target and source
+   * of junction object. If no agent is present within the context, apply target and source condition to user.
+   */
+  public boolean userAgentAuthorization(X x, UserUserJunction junctionObj, String permission){
+    User user = (User) x.get("user");
+    User agent = (User) x.get("agent");
+    AuthService auth = (AuthService) x.get("auth");
+
+    if ( user == null ) {
+      throw new AuthenticationException();
+    }
+
+    // Check agent or user to authorize the request as.
+    User authorizedUser = agent != null ? agent : user;
+
+    // Check junction object relation to authorized user.
+    boolean authorized =
+        ( SafetyUtil.equals(junctionObj.getTargetId(), authorizedUser.getId()) ||
+        SafetyUtil.equals(junctionObj.getSourceId(), authorizedUser.getId()) );
+
+    return ( auth.check(x, permission) && agent == null ) || authorized;
```

Comment:
```suggestion
    return authorized || auth.check(x, permission);
```

Old:
If I'm acting as someone and the agent is target or source: true
If I'm acting as someone and the agent is not target or source: false
If I'm not acting as someone and the user is target or source: true
If I'm not acting as someone and the user is not target or source: Permission check on user's group

New:
If I'm acting as someone and the agent is target or source: true
If I'm acting as someone and the agent is not target or source: Permission check on junction group
If I'm not acting as someone and the user is target or source: true
If I'm not acting as someone and the user is not target or source: Permission check on user's group